### PR TITLE
Update modification date, when modifying element

### PR DESCRIPTION
**Pull request by souissimai**

### CODE REVIEW

File: src/main/java/org/gridsuite/directory/server/repository/DirectoryElementEntity.java [hunk 1]
```diff
@@ -68,21 +69,33 @@ public class DirectoryElementEntity {
...
+        if (isDescriptionUpdated || isElementNameUpdated) {
+            updateModificationAttributes(lastModifiedBy, LocalDateTime.now(ZoneOffset.UTC));
+        }
         return this;
     }
 
+    public void updateModificationAttributes(String lastModifiedBy,
+                                             LocalDateTime lastModificationDate) {
+        this.setLastModificationDate(lastModificationDate);
+        if (Objects.nonNull(lastModifiedBy)) {
```
Comment (YenguiSeddik) [inline src/main/java/org/gridsuite/directory/server/repository/DirectoryElementEntity.java:94]: can the userId be null ?

**Review by YenguiSeddik — APPROVED** [review]

File: src/test/java/org/gridsuite/directory/server/DirectoryTest.java [hunk 2]
```diff
@@ -166,32 +168,8 @@ public void test() throws Exception {
         // Rename the root directory
         renameElement(uuidNewDirectory, uuidNewDirectory, "userId", "newName", true, false);
 
-        checkRootDirectoriesList("userId", List.of(toElementAttributes(uuidNewDirectory, "newName", DIRECTORY, false, "userId", null, creationDateNewDirectory, creationDateNewDirectory, "userId")));
+        checkRootDirectoriesList("userId", List.of(toElementAttributes(uuidNewDirectory, "newName", DIRECTORY, false, "userId", null, creationDateNewDirectory, modificationDateNewDirectory, "userId")));
 
-        // Change root directory access rights public => private
```
[the remark this answers is not in the document]
Comment (souissimai) [reply src/test/java/org/gridsuite/directory/server/DirectoryTest.java:174]: I put it again

File: src/main/java/org/gridsuite/directory/server/repository/DirectoryElementEntity.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (souissimai) [reply src/main/java/org/gridsuite/directory/server/repository/DirectoryElementEntity.java:94]: I dont think so, will remove this check

[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/directory/server/repository/DirectoryElementEntity.java]: Add @NonNull

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**